src/pages/Forever/useRedeemCode.ts

```diff
@@ -0,0 +1,81 @@
+import { useCurrentLocale } from 'components/utils/CurrentLocale'
+import { useRedeemCodeV2Mutation } from 'data/graphql'
+import { FormikHelpers } from 'formik'
+import { useHistory } from 'react-router'
+import { captureSentryError } from 'utils/sentry-client'
+import { RedeemCodeFormValue } from './components/RedeemCode'
+
+export const useRedeemCode = () => {
+  const [redeemCode] = useRedeemCodeV2Mutation()
+  const locale = useCurrentLocale()
+  const history = useHistory()
+
+  const handleSubmit = async (
+    form: RedeemCodeFormValue,
+    actions: FormikHelpers<RedeemCodeFormValue>,
+  ) => {
+    try {
+      const result = await redeemCode({ variables: { code: form.code } })
+      if (!result) {
+        return
+      }
+
+      if (result.data?.redeemCodeV2.__typename === 'SuccessfulRedeemResult') {
+        // TODO redirect to correct success screens
+        history.push(`/${locale}/new-member`)
+        return
+      }
+      if (result.data?.redeemCodeV2.__typename === 'CannotRedeemOwnCampaign') {
+        actions.setErrors({ code: 'FOREVER_ERROR_CANNOT_REDEEM_OWN_CAMPAIGN' })
+        return
+      }
+      if (
+        result.data?.redeemCodeV2.__typename ===
+        'CampaignCannotBeCombinedWithExisting'
+      ) {
+        actions.setErrors({
+          code: 'FOREVER_ERROR_CAMPAIGN_CANNOT_BE_COMBINED_WITH_EXISTING',
+        })
+        return
+      }
+      if (result.data?.redeemCodeV2.__typename === 'CampaignHasExpired') {
+        actions.setErrors({ code: 'FOREVER_ERROR_CAMPAIGN_HAS_EXPIRED' })
+        return
+      }
+      if (
+        result.data?.redeemCodeV2.__typename ===
+        'MemberIsNotEligibleForCampaign'
+      ) {
+        actions.setErrors({
+          code: 'FOREVER_ERROR_CAMPAIGN_CANNOT_BE_COMBINED_WITH_EXISTING',
+        })
+        return
+      }
+
+      if (result.data?.redeemCodeV2.__typename === 'CampaignDoesNotExist') {
+        actions.setErrors({ code: 'FOREVER_ERROR_CAMPAIGN_DOES_NOT_EXIST' })
```

Comment:
we could make this a switch statement instead, right?

Reply:
very much so! just made it simple and ugly to start with :P